lightning/src/ln/channelmanager.rs

```diff
@@ -11038,6 +11038,14 @@ where
 
 		match message {
 			OffersMessage::InvoiceRequest(invoice_request) => {
+				if self.default_configuration.manually_handle_bolt12_messages {
+					let event = Event::InvoiceRequestReceived {
+						invoice_request, context, responder,
+					};
+					self.pending_events.lock().unwrap().push_back((event, None));
+					return None;
+				}
```

Comment:
We should only create an event if the invoice request can be verified.

Reply:
Makes sense!
Addressed the point in [pr3318.03](https://github.com/shaavan/rust-lightning/commits/pr3318.03)
Thanks a lot for the pointer!